### PR TITLE
po/fix masks manager

### PR DESCRIPTION
**Pull request by TurboGit**
Two fixes here (For #13755).
- make circle/ellipsis/gradient modifiable after being created in mask manager
- fix crash for gradien when created in mask manager

There is another issue I have found, the brush is not visible when after being created from mask manager. I have tracked this down to:

```
commit dbc3f4f15f498c8bdeb25a16c581822acb645a51
Author: Diederik ter Rahe <dterrahe@yahoo.com>
Date:   Sun Oct 9 16:44:57 2022 -0400

    switch dt_lib_masks_t.gui_reset to standard darktable.gui->reset

 src/develop/develop.c        |  6 ++--
 src/develop/develop.h        |  4 +--
 src/develop/masks/brush.c    |  4 +--
 src/develop/masks/circle.c   |  4 +--
 src/develop/masks/ellipse.c  |  4 +--
 src/develop/masks/gradient.c |  4 +--
 src/develop/masks/masks.c    | 14 +++------
 src/develop/masks/path.c     |  4 +--
 src/libs/masks.c             | 73 +++++++++++++++++++++-----------------------
 9 files changed, 52 insertions(+), 65 deletions(-)
```

@dterrahe : Can you help for this one? I have hard time to see what is wrong... TIA.

### CODE REVIEW

File: src/develop/masks/path.c [hunk 1]
```diff
@@ -1339,7 +1339,7 @@ static int _path_events_button_pressed(struct dt_iop_module_t *module,
       dt_dev_masks_selection_change(darktable.develop, crea_module, form->formid);
       gui->creation_module = NULL;
 
-      if(gui->creation_continuous)
+      if(crea_module && gui->creation_continuous)
```
Comment (dterrahe) [inline src/develop/masks/path.c:1342]: @TurboGit I thought it made more sense to have the check for `crea_module` in the next if statement

File: src/develop/masks/path.c [hunk 2]
```diff
@@ -1339,7 +1339,7 @@ static int _path_events_button_pressed(struct dt_iop_module_t *module,
...
       {
         //spot and retouch manage creation_continuous in their own way
         if(strcmp(crea_module->so->op, "spots") != 0
```
Comment (dterrahe) [inline src/develop/masks/path.c:1345]: So
```
      if(crea_module
         && strcmp(crea_module->so->op, "spots") != 0
         && strcmp(crea_module->so->op, "retouch") != 0)
```

There's s identical code that needs the same fix in brush.c

Comment (TurboGit) [reply src/develop/masks/path.c:1346]: Yep! Now fixed.

File: src/develop/masks/path.c (hunk 1 above)
Comment (TurboGit) [reply src/develop/masks/path.c]: But this will enable continuous creation of path in mask manager which is not supported for circles & ellipses.

**Comment by dterrahe** [comment]
The problem with newly created widgets is caused by
https://github.com/darktable-org/darktable/blob/4e4e7b65d7676ad5e2d7901f2ac585fed8964ee3/src/libs/masks.c#L877
in `_tree_selection_change` which should _probably_ move to `_update_all_properties`

But this creates a new set of problems so I'm looking into fixing those...

File: src/develop/masks/path.c (hunk 1 above)
Comment (dterrahe) [reply src/develop/masks/path.c]: Shouldn't that work though? (i.e. shouldn't the fix be with circles & ellipses?)

Comment (TurboGit) [reply src/develop/masks/path.c]: Ok, it works for circle, ellipse, path & gradient now. I'll do the same for brush when your fixes will be in place.

**Comment by dterrahe** [comment]
I've submitted a PR to your PR here https://github.com/TurboGit/darktable/pull/3

The trickiness of this problem is that we don't want the mask manager list to update too often, because it can interrupt other operations, but we also want the property sliders in the collapsible section to be always up to date. I think I managed to get both working, but you'll want to exercise _all_ functionality of the mask manager (and of the mask sections of modules) to see if it works in groups, in modules and stand alone.

I _think_ move up/down also works, but please also check undo and if all the different modes update correctly after a move (I vaguely remember an issue about those needing amending when getting to the top of a group so there's probably still outstanding work here).

Groups still have a tendency to collapse too often when trying to issue an operation, but I don't know if this maybe always was the case.

**Comment by TurboGit** [comment]
@dterrahe : Thanks a lot! I have now integrated your patch, added support for brush continuous creation and reformatted all the masks code. If the CI is clear I'll merge all this has it is definitely a good step forward. The mask manager is again usable.

**Pull request merged**